tensorboard/loader.py

```diff
@@ -399,6 +401,254 @@ def __str__(self):
+          frequently than window.
+      clock: Function returning a float with the number of seconds
+          since the UNIX epoch in zulu time.
+
+    :type window: float
+    :type clock: () -> float
+    """
+    self._window = window
+    self._clock = clock
+    self._points = collections.deque()
+    self._last_value = None  # type: int
+    self._last_time = None  # type: float
+
+  def get_rate(self):
+    """Determines rate of increase in value per second averaged over window.
+
+    Returns:
+      An integer representing the rate or None if not enough
+      information has been collected yet.
+
+    :rtype: int
+    """
+    toto = 0.0
+    points = []
+    for i, (rate, weight, _) in enumerate(self._points):
+      weight *= 1.0 / (i + 1)
+      toto += weight
+      points.append((rate, weight))
+    if not toto:
+      return 0
+    return int(sum(w / toto * r for r, w in points))
+
+  def set_value(self, value):
+    """Sets number state.
+
+    This method adds a delta between value and the value of the last
+    time this method was called. Therefore the first invocation does
+    not add a delta.
+
+    Raises:
+      ValueError: If value is less than the last value.
+
+    :type value: int
+    """
+    now = self._clock()
+    if self._last_value is None:
+      self._last_value = value
+      self._last_time = now
+      return
+    if value < self._last_value:
+      raise ValueError('%d < %d' % (value, self._last_value))
+    delta = value - self._last_value
+    elapsed = now - self._last_time
+    if not elapsed:
+      return
+    self._points.appendleft((delta / elapsed, elapsed, now))
+    self._last_time = now
+    self._last_value = value
+    self._remove_old_points()
+
+  def bump(self):
+    """Makes time since last set_value count for nothing."""
+    self._last_time = self._clock()
+
+  def _remove_old_points(self):
+    threshold = self._clock() - self._window
+    while self._points:
+      r, e, t = self._points.pop()
+      if t > threshold:
+        self._points.append((r, e, t))
+        break
+
+
+@util.closeable
+class Progress(object):
+  """Terminal UI for displaying job progress in terms of bytes.
+
+  On teletypes, this class will display a nice ephemeral unicode
+  progress bar. Otherwise it just emits periodic log messages.
+
+  This class keeps track of the rate at which input is processed, as
+  well as the rate it grows. These values are represented to the user
+  using the DELTA and NAMBLA symbols.
+
+  An alarm is displayed if the consumption rate falls behind the
+  production rate. In order for this to be calculated properly, the
+  sleep method of this class should be used rather than time.sleep.
+  """
+
+  BAR_INTERVAL_SECONDS = 0.25
+  BAR_LOGGER = logging.getLogger('tensorflow' + util.LogHandler.EPHEMERAL)
+  BAR_WIDTH = 45
+  BLOCK_DARK = u'\u2593'
+  BLOCK_LIGHT = u'\u2591'
+  DELTA = u'\u2206'
+  LOG_INTERVAL_SECONDS = 5.0
+  NABLA = u'\u2207'
```

Comment:
Any reason not to write `BLOCK_DARK = u'▓'`, etc., per [the style guide](https://engdoc.corp.google.com/eng/doc/devguide/java/styleguide.shtml?cl=head#s2.3.3-non-ascii-characters)?